esx_service/utils/auth.py

```diff
@@ -0,0 +1,360 @@
+    global _auth_mgr
+
+    try:
+        cur = _auth_mgr.conn.execute("SELECT name FROM sqlite_master WHERE type = 'table' and name = 'tenants';")
+        result = cur.fetchall()
+    except sqlite3.Error, e:
+        logging.error("Error %s when checking whether table tenants exists or not", e)
+        return str(e), False
+
+    if not result:
+        logging.error("table tenants does not exist")
+        error_info = "table tenants does not exist"
+        return error_info, False
+    
+    try:
+        cur = _auth_mgr.conn.execute("SELECT name FROM sqlite_master WHERE type = 'table' and name = 'vms';")
+        result = cur.fetchall()
+    except sqlite3.Error, e:
+        logging.error("Error %s when checking whether table vms exists or not", e)
+        return str(e), False
+
+    if not result:
+        logging.error("table vms does not exist")
+        error_info = "table vms does not exist"
+        return error_info, False
+
+    try:
+        cur = _auth_mgr.conn.execute("SELECT name FROM sqlite_master WHERE type = 'table' and name = 'privileges';")
+        result = cur.fetchall()
+    except sqlite3.Error, e:
+        logging.error("Error %s when checking whether table privileges exists or not", e)
+        return str(e), False
+
+    if not result:
+        logging.error("table privileges does not exist")
+        error_info = "table privileges does not exist"
+        return error_info, False
+    
+    try:
+        cur = _auth_mgr.conn.execute("SELECT name FROM sqlite_master WHERE type = 'table' and name = 'volumes';")
+        result = cur.fetchall()
+    except sqlite3.Error, e:
+        logging.error("Error %s when checking whether table volumes exists or not", e)
+        return str(e), False
+
+    if not result:
+        logging.error("table volumes does not exist")
+        error_info = "table volumes does not exist"
+        return error_info, False
+    
+    return None, True
+
+def authorize(vm_uuid, datastore, cmd, opts):
+    """ Check whether the command can be run on this VM.
+
+        Return value: result, tenant_uuid, tenant_name
+
+        - result: return None if the command can be run on this VM, otherwise, return 
+        corresponding error message
+        - tenant_uuid: If the VM belongs to a tenant, return tenant_uuid, otherwise, return
+        None
+        - tenant_name: If the VM belongs to a tenant, return tenant_name, otherwise, return 
+        None
+
+    """
+    logging.debug("Authorize: vm_uuid=%s", vm_uuid)
+    logging.debug("Authorize: datastore=%s", datastore)
+    logging.debug("Authorize: cmd=%s", cmd)
+    logging.debug("Authorize: opt=%s", opts)
+
+    connect_auth_db()
+
+    # If table "tenants", "vms", "privileges" or "volumes" does not exist
+    # don't need auth check
+    if not tables_exist():
+        logging.error("Required tables in auth db do not exist")
+        error_info = "Required tables in aut db do not exist"
```

Comment:
I think we should not allow the operation. If auth DB is present, then the required table should be created. Otherwise, it means something is wrong.